crates/tabby-inference/src/code.rs

```diff
@@ -31,10 +31,20 @@ pub struct CodeGeneration {
 }
 
 impl CodeGeneration {
-    pub fn new(imp: Arc<dyn CompletionStream>) -> Self {
+    pub fn new(imp: Arc<dyn CompletionStream>, config: Option<ModelConfig>) -> Self {
+        let stop_condition_factory = match config {
+            Some(ModelConfig::Local(config)) => StopConditionFactory::with_stop_words(
+                config.additional_stop_words.unwrap_or_default(),
+            ),
+            Some(ModelConfig::Http(config)) => StopConditionFactory::with_stop_words(
+                config.additional_stop_words.unwrap_or_default(),
+            ),
+            _ => StopConditionFactory::default(),
+        };
+
```

Comment:
```suggestion
        let additional_stop_words = match config {
            Some(ModelConfig::Local(config)) =>  config.additional_stop_words.unwrap_or_default(),
            ...
        };

```

Reply:
the `stop_condition_factory` here means all the stop words, not only the additional stop words, are we really want to rename the `stop_condition_factory` to `additional_stop_words`?

Reply:
I'm asking a restructure, rather than a renaming.
```rust

        let additional_stop_words = match config {
            Some(ModelConfig::Local(config)) =>  config.additional_stop_words.unwrap_or_default(),
            ...
        };

       let stop_condition_factory = StopConditionFactory::with_stop_words(additional_stop_words).
```

So you don't need have to write `StopConditionFactory::with_stop_words` multiple times.